pycodestyle.py

```diff
@@ -1509,18 +1512,28 @@ def ambiguous_identifier(logical_line, tokens):
     E741: def foo(l=12):
     E741: l = foo(l=12)
     E741: for l in range(10):
+    E741: [l for l in lines if l]
+    E741: lambda l: None
+    E741: lambda a=x[1:5], l: None
+    E741: lambda **l: None
     E742: class I(object):
     E743: def l(x):
     """
     is_func_def = False  # Set to true if 'def' is found
+    is_lambda_def = False
     parameter_parentheses_level = 0
     idents_to_avoid = ('l', 'O', 'I')
     prev_type, prev_text, prev_start, prev_end, __ = tokens[0]
-    for token_type, text, start, end, line in tokens[1:]:
+    for index in range(1, len(tokens)):
+        token_type, text, start, end, line = tokens[index]
+        next_token = tokens[index + 1] if index < len(tokens) - 1 else None
+        next_text = next_token[1] if next_token else None
```

Comment:
idk if it makes much performance difference here -- but I suspect that these should be inside the `if` below such that they aren't computed unnecessarily -- `pycodestyle`'s checks are fairly "hot" in terms of execution time